tests/cupy_tests/core_tests/test_ndarray_reduction.py

```diff
@@ -228,6 +231,26 @@ def test_cub_min(self, xp, dtype, axis):
             a = xp.ascontiguousarray(a)
         elif self.order in ('f', 'F'):
             a = xp.asfortranarray(a)
+
+        if xp is numpy:
+            return a.min(axis=axis)
+
+        # xp is cupy, first ensure we really use CUB
+        ret = cupy.empty(())  # Cython checks return type, need to fool it
+        full_func = 'cupy.core._routines_statistics.cub.device_reduce'
+        full_patch = mock.patch(full_func, return_value=ret)
+        seg_func = 'cupy.core._routines_statistics.cub.device_segmented_reduce'
+        seg_patch = mock.patch(seg_func, return_value=ret)
+        with full_patch as f, seg_patch as s:
+            assert f.call_count == s.call_count == 0
+            a.min(axis=axis)
+            if len(axis) == len(self.shape):
+                assert f.call_count == 1
+                assert s.call_count == 0
+            else:
+                assert f.call_count == 0
+                assert s.call_count == 1
+        # ...then perform the actual computation
```

Comment:
I think this can be simplified with 
```python
if len(axis) == len(self.shape):
   func = 'cupy.core._routines_statistics.cub.device_reduce'
else:
   func = 'cupy.core._routines_statistics.cub.device_segmented_reduce'
patch = mock.patch(func, return_value=ret)

with full_patch as f:
     assert f.call_count == 0
     a.min(axis=axis)
     assert f.call_count == 1
```

Reply:
Moreover, you can define your own context manager to do it and avoid code repetition
in all the tests.
```
with check_called(patch):
   myfunc(x)
```

Reply:
Very neat! Will do.